### PR TITLE
Make searchable snapshot indexes read-only but allow deletion (#4764)

### PR DESCRIPTION
**Pull request by Vishalks**
This commit adds a new cluster block that is added by default to all searchable snapshot indexes to block writes and metadata changes.

Signed-off-by: Vishal Sarda <vsarda@amazon.com>
(cherry picked from commit c4fc1bc0f29bc7a005e4755be5d2e1b22b7de72a)
Signed-off-by: Vishal Sarda <vsarda@amazon.com>

### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Introduce experimental searchable snapshot API ([#4680](https://github.com/opensearch-project/OpenSearch/pull/4680))
 - Introduce Remote translog feature flag([#4158](https://github.com/opensearch-project/OpenSearch/pull/4158))
 - Add groupId value propagation tests for ZIP publication task ([#4848](https://github.com/opensearch-project/OpenSearch/pull/4848))
+- Recommissioning of zone. REST layer support. ([#4624](https://github.com/opensearch-project/OpenSearch/pull/4604))
+- Make searchable snapshot indexes read-only but allow deletion ([#4764](https://github.com/opensearch-project/OpenSearch/pull/4764))
```
Comment (andrross) [inline CHANGELOG.md:23]: This line should be the only change in the this file

Comment (Vishalks) [reply CHANGELOG.md:23]: Agreed and I had the same thought when following the backporting instructions on my original PR. It came after I cherry picked my original PR commit
image

Comment (andrross) [reply CHANGELOG.md:23]: Yeah you should just pick the line of the change you're backporting. There's a lot of contention on the CHANGELOG file so this is tedious, but hopefully we can improve this process.

Comment (Vishalks) [reply CHANGELOG.md:22]: Updated

**Review by andrross — APPROVED** [review]

**Pull request merged**